src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -1,14 +1,146 @@
 package cromwell.binding
 
+import cromwell.binding.AstTools.AstNodeName
+import cromwell.parser.WdlParser.{Ast, AstList}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+
+object Scope {
+
+  /**
+   * Generate a Scope Tree from an Ast.
+   * Ast must match a Scope (currently: Workflow, Scatter or Call)
+   * @throws UnsupportedOperationException if node does not match any supported Scopes
+   * @return generated Scope Tree.
+   */
+  def generateTree(node: Ast, namespaces: Seq[WdlNamespace], tasks: Seq[Task], wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Scope = {
+
+    val scopeIndexes: scala.collection.mutable.Map[Class[_ <: Scope], Int] = scala.collection.mutable.HashMap.empty
```

Comment:
Yep, I kept it like that to make it clearly visible that the map is mutable but we can import it if you think it's fine.